src/commands/admin/activity.ts

```diff
@@ -9,19 +9,23 @@ export async function permissions(msg: Message): Promise<boolean> {
 	return await botMasterPerms(msg);
 }
 
-export async function run(msg: Message, [type, content]: [ActivityType, string]): Promise<Message> {
+export async function run(msg: Message, [type, name]: [ActivityType, string]): Promise<Message> {
 	const bot = msg.client;
-	bot.user.setActivity(`${content}`, { type: type });
-	return msg.channel.send(`set ${BOT.NAME}'s activity to ${type} ${content}`);
+	bot.user.setActivity(name, { type });
+	bot.mongo.collection(DB.CLIENT_DATA).updateOne(
+		{ _id: bot.user.id },
+		{ $set: { status: { type, name } } },
+		{ upsert: true });
```

Comment:
What's upsert? On a quick Ctrl-F, I didn't see any checks for it.

Reply:
`upsert: true` basically makes a new document if it couldn't find one matching the query.